File: server/th_tft_display.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <sqlite3.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>

static int callback(void *NotUsed, int argc, char **argv, char **azColName){
  time_t theTime = time(NULL);
  struct tm *aTime = localtime(&theTime);
  char command[100];
  sprintf(command, "%s_%s_%s", argv[1], argv[0], argv[2]);
  printf("%s \n", command);

  int fd;
  fd = open("/dev/ttyACM0", O_RDWR | O_NOCTTY | O_NDELAY);
  write(fd, command, sizeof(command));

  return 0;
}

int main(int argc, char* argv[])
{
  sqlite3 *db;
  char *zErrMsg = 0;
  int  rc;
  const char *sql;

  while(true)
  {
    /* Open database */
    rc = sqlite3_open("/home/pi/workspace/raspi_net/server/db/sys.db", &db);
    if( rc ){
      fprintf(stderr, "Can't open database: %s\n", sqlite3_errmsg(db));
      exit(0);
    }else{
      fprintf(stdout, "Opened database successfully\n");
    }

    sql = "SELECT TEMP, HUMI, strftime('%H',CREATED) FROM TEMP_HUMI ORDER BY ID DESC LIMIT 1";

    /* Execute SQL statement */
    rc = sqlite3_exec(db, sql, callback, 0, &zErrMsg);
    if( rc != SQLITE_OK ){
      fprintf(stderr, "SQL error: %s\n", zErrMsg);
      sqlite3_free(zErrMsg);
    }else{
      fprintf(stdout, "Send data successfully\n");
    }
    sqlite3_close(db);

//sleep(60*60);
    sleep(10);
  }
    return 0;
}

File: server/th_tft_display.py
#!/usr/bin/python
import serial, sqlite3, time
#import os, sqlite3, time
#from serial.serialutil import to_bytes

#d = to_bytes(command)
#serial = os.open('/dev/ttyACM0', os.O_RDWR | os.O_NOCTTY | os.O_NONBLOCK)
#os.write(serial, d)
ser = serial.Serial('/dev/ttyACM0', 9600, timeout=1);
while 1:
  time.sleep(3);
  conn = sqlite3.connect('/home/pi/workspace/raspi_net/server/db/sys.db')
  c = conn.cursor()
  print "Opened database successfully";

  cursor = c.execute("SELECT TEMP, HUMI FROM TEMP_HUMI ORDER BY ID DESC LIMIT 1")
  row = cursor.fetchone()
  command = str(round(row[1],2)) + "_" + str(round(row[0],2)) + "_" + str(time.strftime("%H", time.localtime()))

  print "Operation done successfully";
  conn.close()

  print command + "\n";

  ser.write(command);
  time.sleep(60*60);
  print(sum);